math/src/stableswap/math.rs

```diff
@@ -183,23 +201,33 @@ pub fn calculate_withdraw_one_asset<const N: u8, const N_Y: u8>(
 		}
 	}
 
-	let y1 = calculate_y::<N_Y>(&reserves_reduced, Balance::try_from(d1).ok()?, amplification)?;
+	let y1 = calculate_y_internal::<Y>(&reserves_reduced, Balance::try_from(d1).ok()?, amplification)?;
+
+	dbg!(y1);
+	dbg!(asset_reserve);
 
 	let dy = asset_reserve.checked_sub(y1)?;
 
 	let dy_0 = reserves[asset_index].checked_sub(y)?;
 
 	let fee = dy_0.checked_sub(dy)?;
 
-	Some((dy, fee))
+	let amount_out = normalize_value(dy, target_precision, asset_out_decimals, Rounding::Down);
+	let fee = normalize_value(fee, target_precision, asset_out_decimals, Rounding::Down);
```

Comment:
Do we need to normalize these values? I think we compute them from normalized values.

Reply:
yes. this converts it back to asset decimals

Reply:
Ah. Now I see.